src/modals/SendModals/SendModal.tsx

```diff
@@ -99,48 +103,66 @@ function SendModal<PT extends { fromAddress: Address }, T extends PT>({
     buildConsolidationTransactions()
   }, [consolidationRequired, transactionData])
 
-  const txContext: TxContext = {
-    setIsSweeping,
-    sweepUnsignedTxs,
-    setSweepUnsignedTxs,
-    setFees,
-    unsignedTransaction,
-    setUnsignedTransaction,
-    unsignedTxId,
-    setUnsignedTxId,
-    isSweeping,
-    consolidationRequired
-  }
-
-  const buildTransactionExtended = async (data: T) => {
-    setTransactionData(data)
-    setIsLoading(true)
+  const txContext: TxContext = useMemo(
+    () => ({
+      setIsSweeping,
+      sweepUnsignedTxs,
+      setSweepUnsignedTxs,
+      setFees,
+      unsignedTransaction,
+      setUnsignedTransaction,
+      unsignedTxId,
+      setUnsignedTxId,
+      isSweeping,
+      consolidationRequired
+    }),
+    [consolidationRequired, isSweeping, sweepUnsignedTxs, unsignedTransaction, unsignedTxId]
+  )
 
-    try {
-      await buildTransaction(data, txContext)
+  const buildTransactionExtended = useCallback(
+    async (data: T) => {
+      setTransactionData(data)
+      setIsLoading(true)
 
-      if (!isConsolidateUTXOsModalVisible) {
-        setStep('info-check')
-      }
-    } catch (e) {
-      // TODO: When API error codes are available, replace this substring check with a proper error code check
-      const { error } = e as APIError
-      if (error?.detail && (error.detail.includes('consolidating') || error.detail.includes('consolidate'))) {
-        setIsConsolidateUTXOsModalVisible(true)
-        setConsolidationRequired(true)
-      } else {
-        dispatch(transactionBuildFailed(getHumanReadableError(e, t('Error while building transaction'))))
+      try {
+        await buildTransaction(data, txContext)
+
+        if (!isConsolidateUTXOsModalVisible) {
+          setStep('info-check')
+        }
+      } catch (e) {
+        // TODO: When API error codes are available, replace this substring check with a proper error code check
+        const { error } = e as APIError
+        if (error?.detail && (error.detail.includes('consolidating') || error.detail.includes('consolidate'))) {
+          setIsConsolidateUTXOsModalVisible(true)
+          setConsolidationRequired(true)
+        } else {
+          dispatch(transactionBuildFailed(getHumanReadableError(e, t('Error while building transaction'))))
+        }
       }
-    }
 
-    setIsLoading(false)
-  }
+      setIsLoading(false)
+    },
+    [buildTransaction, dispatch, isConsolidateUTXOsModalVisible, t, txContext]
+  )
+
+  useEffect(() => {
+    if (initialStep === 'info-check' && transactionData) {
+      buildTransactionExtended(transactionData)
+    }
+  }, [buildTransactionExtended, initialStep, transactionData])
```

Comment:
This was needed so that we can skip the build step and go directly to the info-check step when receiving a WC request with predefined tx data (ex: when swapping).